### ISSUE TITLE
Azure Monitor plugin: Link from Prometheus exemplars to App Insights fails

### ISSUE THREAD

**Issue by kayodeprinceMS** [grafana/grafana#90011]
### What happened?

Testing on 11.2 
Prometheus and Azure Monitor datasources configured.
Exemplars and traces are working fine independently but the automatic integration via Exemplar link is not. 
 
![image](https://github.com/grafana/grafana/assets/58087222/5203aa54-357a-4de2-bf5d-6d0e280c3a99)

![image](https://github.com/grafana/grafana/assets/58087222/36dd2e1b-100e-4d12-993a-11fcd68f5aad)

The issue may be related to the Azure Monitor datasource not saving the default subscription because of the Log Analytics warning shown below. Or if the default subscription is not saved for any other reason.

![image](https://github.com/grafana/grafana/assets/58087222/8e17d4bb-870b-4da9-a3e9-726f04e170a3)

 

### What did you expect to happen?

Trace view of App Insights data to load successfully.

### Did this work before?

No, new feature.

### How do we reproduce it?

1. Create Azure Monitor plugin where default subscription is not selected or is not saved because of data source warning for e.g. no Log Analytics endpoint found.
2. Configure Prometheus plugin Exemplar to point to Azure Monitor plugin 
3. Click on App Insights link for Exemplar

### Is the bug inside a dashboard panel?

Error occurs in Explore and dashboard panels

| Key | Value |
|--|--|
| Panel | traces @ 11.2.0-185387 |
| Grafana | Grafana v11.2.0-185387 (fce03cd724) // Open Source |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": []
        }
      ],
      "type": "traces",
      "title": "Reproduced with embedded data",
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "pluginVersion": "11.2.0-185387"
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >traces @ 11.2.0-185387</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[grafana-azure-monitor-datasource]</td>\n    </tr>\n    \n    <tr><th>Data</th><td>Error 0 frames, 0 fields, 0 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>Grafana v11.2.0-185387 (fce03cd724) // Open Source</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"datasource\": {\n    \"type\": \"grafana-azure-monitor-datasource\",\n    \"uid\": \"bdqkhwxqyj668c\"\n  },\n  \"targets\": [\n    {\n      \"query\": \"1a251b70e266e101b1796c65659caafa\",\n      \"queryType\": \"traceql\",\n      \"refId\": \"A\",\n      \"azureTraces\": {\n        \"operationId\": \"1a251b70e266e101b1796c65659caafa\",\n        \"resultFormat\": \"trace\",\n        \"traceTypes\": [\n          \"availabilityResults\",\n          \"dependencies\",\n          \"customEvents\",\n          \"exceptions\",\n          \"pageViews\",\n          \"requests\",\n          \"traces\"\n        ]\n      }\n    }\n  ],\n  \"type\": \"traces\",\n  \"title\": \"New Panel\",\n  \"gridPos\": {\n    \"x\": 0,\n    \"y\": 0,\n    \"w\": 12,\n    \"h\": 8\n  },\n  \"id\": 1,\n  \"pluginVersion\": \"11.2.0-185387\"\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: New Panel // 2024-07-03 15:00:06",
  "tags": [
    "debug",
    "debug-traces"
  ],
  "time": {
    "from": "2024-07-03T12:58:29.668Z",
    "to": "2024-07-03T18:58:29.668Z"
  }
}
```
</details>

### Environment (with versions)?

Grafana: 11.2
OS: Windows
Browser: Chrome and IE

### Grafana platform?

Docker

### Datasource(s)?

Azure Monitor

**Comment by kayodeprinceMS** [comment]
FYI @taleena for future tagging

**Comment by jcolladokuri** [comment]
Did some quick investigation, this might be related to the fact that no default subscription is loaded for a datasource if `Load Subscriptions` is not clicked, this seems to be happening independent of whether the resource health check succeeds or fails.

**Comment by aangelisc** [comment]
Hi all,

I concur what @jcolladokuri has discovered and I have a fix up in #90185

**Comment by aangelisc** [comment]
This should be fixed and released now. Closing this issue but please re-open if the issue occurs again!
